src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -828,6 +828,20 @@ internal static X509Certificate2 CreateSelfSignedCertificate(
             request.CertificateExtensions.Add(extension);
         }
 
+        // Only add the SKI and AKI extensions if neither is already present.
+        // OpenSSL needs these to correctly identify the trust chain for a private key. If multiple certificates don't have a subject key identifier and share the same subject,
+        // the wrong certificate can be chosen for the trust chain, leading to validation errors.
+        if (!request.CertificateExtensions.OfType<X509SubjectKeyIdentifierExtension>().Any() && !request.CertificateExtensions.OfType<X509AuthorityKeyIdentifierExtension>().Any())
```

Comment:
This isn't _quite_ right, because someone could include `X509Extension` without specializing it to the types listed.

If you want to be defensive against them being added, you need to check something more like

```suggestion
        if (request.CertificateExtensions.Count(ext => ext.Oid.Value is "2.5.29.14" or "2.5.29.35") == 0)
```

And, of course, I feel compelled to ask "can you add a test for this?"